src/app/context/wallet/CosmosWalletProvider.tsx

```diff
@@ -58,6 +63,15 @@ export const CosmosWalletProvider = ({ children }: PropsWithChildren) => {
       const client = await providers.cosmosProvider.getSigningStargateClient({
         registry: getBbnRegistry(),
       });
+
+      // TODO: Temporary solution until #409 is merged
+      // Create QueryClient directly
+      const tmClient = await Tendermint34Client.connect(
+        "https://rpc.devnet.babylonlabs.io",
+      );
+      const queryClient = QueryClient.withExtensions(tmClient);
```

Comment:
@jrwbabylonlab does it mean that this PR is in Draft/WIP state atm?

Reply:
@jrwbabylonlab I don't think that instances of QueryClient and SigningStargetClient should be inside CosmosWalletProvider. Instead CosmosWalletProvider just provides factory methods to create such instances somewhere else in the code. CosmosWalletProvider is just a thin adapter for Tomo/Native wallet's network providers to avoid tight coupling between DApp and current wallet provider itself